app/components/views/Offset/index.tsx

```diff
@@ -125,6 +126,32 @@ export const Offset = (props: Props) => {
     dispatch(setAppState({ notificationStatus: { statusType, message } }));
   };
 
+  /** Initialize input from params after they are extracted, validated & stripped */
+  useEffect(() => {
+    if (params?.inputToken) {
+      setSelectedInputToken(params.inputToken);
+    }
+    if (params?.retirementToken) {
+      setSelectedRetirementToken(params.retirementToken);
+    }
+    if (params?.message) {
+      setRetirementMessage(params.message);
+    }
+    if (params?.beneficiary) {
+      setBeneficiary(params.beneficiary);
+    }
+    if (params?.beneficiaryAddress) {
+      setBeneficiaryAddress(params.beneficiaryAddress);
+    }
+    if (params?.projectTokens) {
+      setSpecificAddresses(params.projectTokens);
+    }
+    if (params?.quantity) {
+      setQuantity(params.quantity);
+      setDebouncedQuantity(params.quantity);
```

Comment:
@0xAeterno We rely on a useEffect to query the RPC / contract to fetch the price quote whenever a relevant value changes
https://github.com/KlimaDAO/klimadao/blob/d362e3f1188139c9b42f9cd2a47754a3fcbbe25d/app/components/views/Offset/index.tsx#L187

But if I did that after every `onChange` or state change, it would rapid fire a bunch of network requests which might not return in the same order. Or might break the fragile RPC.

So I mitigate that by `setting quantity => after delay, set debouncedQuantity  => debounceQuantity` triggers the use effect.

We need both `quantity` and `debounceQuantity`. The former because the input is a controlled component (`value={quantity}` and the latter for the useEffect

Reply:
Curious if you could think of a more straightforward way to handle this. Maybe your fancy form libraries cover this scenario somehow?